src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/SynthesizedSymbols/AnonymousType.DelegateTemplateSymbol.cs

```diff
@@ -0,0 +1,174 @@
+                        typeParameters.Add(new AnonymousTypeManager.AnonymousTypeParameterSymbol(containingType, i, "T" + (i + 1)));
+                    }
+
+                    if (!returnsVoid)
+                    {
+                        typeParameters.Add(new AnonymousTypeManager.AnonymousTypeParameterSymbol(containingType, parameterCount, "TResult"));
+                    }
+
+                    return typeParameters.ToImmutableAndFree();
+                }
+            }
+
+            internal AnonymousDelegateTemplateSymbol(AnonymousTypeManager manager, AnonymousTypeDescriptor typeDescr, ImmutableArray<TypeParameterSymbol> typeParametersToSubstitute)
+                : base(manager, typeDescr.Location)
+            {
+                // AnonymousTypeOrDelegateComparer requires an actual location.
+                Debug.Assert(SmallestLocation != null);
+                Debug.Assert(SmallestLocation != Location.None);
+
+                IsParameterizedDelegateType = false;
+
+                TypeMap typeMap;
+                int typeParameterCount = typeParametersToSubstitute.Length;
+                if (typeParameterCount == 0)
+                {
+                    TypeParameters = ImmutableArray<TypeParameterSymbol>.Empty;
+                    typeMap = TypeMap.Empty;
+                }
+                else
+                {
+                    var typeParameters = ArrayBuilder<TypeParameterSymbol>.GetInstance(typeParameterCount);
+                    for (int i = 0; i < typeParameterCount; i++)
+                    {
+                        typeParameters.Add(new AnonymousTypeParameterSymbol(this, i, "T" + (i + 1)));
+                    }
+                    TypeParameters = typeParameters.ToImmutableAndFree();
+                    typeMap = new TypeMap(typeParametersToSubstitute, TypeParameters, allowAlpha: true);
+                }
+
+                var constructor = new SynthesizedDelegateConstructor(this, manager.System_Object, manager.System_IntPtr);
+                // https://github.com/dotnet/roslyn/issues/56808: Synthesized delegates should include BeginInvoke() and EndInvoke().
+                var invokeMethod = createInvokeMethod(this, typeDescr.Fields, typeMap);
+                _members = ImmutableArray.Create<Symbol>(constructor, invokeMethod);
+
+                static SynthesizedDelegateInvokeMethod createInvokeMethod(
+                    AnonymousDelegateTemplateSymbol containingType,
+                    ImmutableArray<AnonymousTypeField> fields,
+                    TypeMap typeMap)
+                {
+                    var parameterCount = fields.Length - 1;
+                    var parameterTypes = ArrayBuilder<TypeWithAnnotations>.GetInstance(parameterCount);
+                    var parameterRefKinds = ArrayBuilder<RefKind>.GetInstance(parameterCount);
+                    for (int i = 0; i < parameterCount; i++)
+                    {
+                        var parameterOrReturn = fields[i];
```

Comment:
>parameterOrReturn [](http://example.com/codeflow?start=28&length=17)

It looks like return is handled separately after this loop. Is the name accurate?